src/Microsoft.TestPlatform.Extensions.HtmlLogger/HtmlLogger.cs

```diff
@@ -335,17 +335,37 @@ private void PopulateHtmlFile()
         ConsoleOutput.Instance.Information(false, htmlFilePathMessage);
     }
 
-    private string GetFilePath(string fileExtension, string fileName)
+    private string GetFilePathAndCreateFile(string fileExtension, string fileName)
     {
         var fullFileFormat = $".{fileExtension}";
-        return Path.Combine(TestResultsDirPath, string.Concat("TestResult_", fileName, fullFileFormat));
+        string fullFilePath;
+        for (short i = 0; i < short.MaxValue; i++)
+        {
+            var fileNameWithIter = i == 0 ? fileName : GetNextIterationFile(fileName, i);
+            fullFilePath = Path.Combine(TestResultsDirPath, string.Concat("TestResult_", fileNameWithIter, fullFileFormat));
+            lock (LockObject)
+            {
+                if (!File.Exists(fullFilePath))
+                {
+                    File.Create(fullFilePath);
+                    return fullFilePath;
+                }
+            }
+        }
+
+        throw new Exception($"Cannot generate unique filename for: {fileName}");
     }
 
     private string FormatDateTimeForRunName(DateTime timeStamp)
     {
         return timeStamp.ToString("yyyyMMdd_HHmmss", DateTimeFormatInfo.InvariantInfo);
     }
 
+    private string GetNextIterationFile(string baseName, short iteration)
+    {
+        return $"{Path.GetFileNameWithoutExtension(baseName)}[{iteration}]{(Path.HasExtension(baseName) ? Path.GetExtension(baseName) : string.Empty)}";
```

Comment:
I like string interpolation but I think it's not bringing much clarity here, maybe we could either add a comment like `// Returns a file name like '<file name>[X].<ext>' or '<file name>[X]' or replace the full interpolation with

```suggestion
        return Path.GetFileNameWithoutExtension(baseName) + $"[{iteration}]" + (Path.HasExtension(baseName) ? Path.GetExtension(baseName) : string.Empty);
```